### ISSUE TITLE
[java] False positive in UnusedAssignment

### ISSUE THREAD

**Issue by davidburstromspotify** [pmd/pmd#2759]
**Affects PMD Version:** 

6.27.0

**Description:**

The [UnusedAssignment](https://pmd.github.io/latest/pmd_rules_java_bestpractices.html#unusedassignment) rule claims the decrement of `a` is unused. The `if` clauses are necessary to provoke the bug. If only

```java
try {
  return somethingThatCanThrowRandomly(1);
} ...
```
is used, there is no false positive.

**Code Sample demonstrating the issue:**

```java
class Test {
    int a() {
        int a = 10; 
        while (a > 0) {
            a--;
            try {
                if (dummy) {
                    return somethingThatCanThrowRandomly(1);
                } else {
                    return somethingThatCanThrowRandomly(2);
                }   
            } catch (RuntimeException e) {
                // retry
            }   
        }
        return 0;
    }   
}

```

**Steps to reproduce:**

`pmd pmd -d srcdir -rulesets category/java/bestpractices.xml`

**Running PMD through:**

CLI / Gradle
